### PR TITLE
QOL-8368 add compatibility with CKAN 2.9

### PR DESCRIPTION
**Pull request by ThrawnCA**

### CODE REVIEW

File: ckanext/ytp/comments/controllers/__init__.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+from ckan.plugins import toolkit
+from ckan.plugins.toolkit import _, abort, c, get_action, h, request, ValidationError
+
+from ckanext.ytp.comments import email_notifications, helpers,\
+    model as comment_model, notification_helpers
+
+
+log = logging.getLogger(__name__)
+
+
```
Comment (duttonw) [inline ckanext/ytp/comments/controllers/__init__.py:18]: might be nice to add comment.

/* Common module functions below */

File: .docker/Dockerfile.ckan [hunk 2]
```diff
@@ -15,13 +16,14 @@ RUN apk add --no-cache curl build-base \
...
 # Install CKAN.
-ENV CKAN_VERSION 2.8.8
 
 RUN . ${VENV_DIR}/bin/activate \
     && pip install setuptools==36.1 \
     && pip install -e "git+https://github.com/ckan/ckan.git@ckan-${CKAN_VERSION}#egg=ckan" \
     && sed -i "s/psycopg2==2.4.5/psycopg2==2.7.7/g" "${VENV_DIR}/src/ckan/requirements.txt" \
-    && pip install -r "${VENV_DIR}/src/ckan/requirements.txt" \
+    && ((test -f "${VENV_DIR}/src/ckan/requirements-py2.txt" && \
+        pip install -r "${VENV_DIR}/src/ckan/requirements-py2.txt") || \
```
Comment (duttonw) [inline .docker/Dockerfile.ckan:25]: should we be loading py2, how do we test for python3 and python2 independently?

File: ckanext/ytp/comments/templates/package/comment_list.html [hunk 3]
```diff
@@ -3,7 +3,8 @@
 {% resource "comments_js/comments.js" %}
 
 {% macro comment_form(values={}, empty=False, hidden=True, prefix="", action="add") %}
-    <form id="{{ prefix + values.id }}" class="form {% if hidden %}hidden{% endif %}" action="{{ pkg_id }}/comments/{% if values.id %}{{ values.id }}/{% endif %}{{ action }}" method="POST">
+    &lt;form id="{{ prefix + values.id }}" class="form comment-{{ action }} {% if hidden %}hidden{% endif %}" action="/{{ content_type }}/{{ pkg_id }}/comments/{% if values.id %}{{ values.id }}/{% endif %}{{ action }}" method="POST"&gt;
```
[the remark this answers is not in the document]
Comment (ThrawnCA) [reply ckanext/ytp/comments/templates/package/comment_list.html:6]: Ah, that's from debugging. I can remove it.

File: ckanext/ytp/comments/plugin/flask_plugin.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class MixinPlugin(p.SingletonPlugin):
+    p.implements(p.IBlueprint)
+    p.implements(p.IClick)
+
+    # IBlueprint
+
+    def get_blueprint(self):
+        return blueprints.get_blueprints()
```
[the remark this answers is not in the document]
Comment (ThrawnCA) [reply ckanext/ytp/comments/plugin/flask_plugin.py:15]: The idea came from Zharktas' work on `ckanext-report`

**Review by duttonw — APPROVED** [review]

**Review by duttonw — APPROVED** [review]

**Review by duttonw — CHANGES_REQUESTED** [review]
is getting there, marking as request changes so i should get a delta between now and final

**Comment by ThrawnCA** [comment]
@duttonw Build is passing now :)

**Review by duttonw — APPROVED** [review]

**Pull request merged**